src/popups/HandicapPopup/HandicapPopupElement.js

```diff
@@ -8,6 +8,121 @@ const propTypes = {
   label: PropTypes.string.isRequired,
 };
 
+// 012 111 22 333
+const simpleTelTester = /([0-9]{1,3}\s?[0-9]{1,3}\s?[0-9]{1,3}\s?[0-9]{1,3})/g;
+// +41 (0) 11 222 33 44
+const complexTelTester = /((([+]{1}[0-9]{1,3})|([+]?[(]{1}[0-9]{1,3}[)]{1})|([(]?[0-9]{4}[)]?))\s{0,4}[)]?[-\s\\.]?[(]?[0-9]{1,4}[)]?([^\r\n][-\s\\.]{0,1}[0-9]{1,3}){1,4})/g;
+
+const emailTester = /[a-zA-Z0-9._+%-]+@[a-zA-Z0-9.-]+[.][a-zA-Z]{2,4}/gm;
+const urlTester = /(www)?[a-z0-9]+([\\-\\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?/g;
+
+const escapeRegExp = string => {
+  return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
+};
+
+const searchMatches = (intialText, testers) => {
+  let matched = [];
+  testers.forEach(t => {
+    const matches = intialText.match(t);
+    if (matches) {
+      matched.push(matches);
+    }
+  });
+  // Flatten array of matches.
+  if (matched.length) {
+    matched = matched.reduce((acc, curr) => [...acc, ...curr], []);
+  }
+
+  if (matched) {
+    // Remove duplicates from match array.
+    return matched.sort().filter((item, pos, ary) => {
+      return !pos || item !== ary[pos - 1];
+    });
+  }
+  return [];
+};
+
+const mailTo = (email, idx) => (
+  <a key={idx} href={`mailto:${email}`}>
+    {email}
+  </a>
+);
+
+const telTo = (telNumber, idx) => (
+  <a key={idx} href={`tel:${telNumber}`}>
+    {telNumber}
+  </a>
+);
+
+const urlHref = (href, idx) => {
+  return (
+    <a
+      key={idx}
+      href={`http://${href}`}
+      rel="noopener noreferrer"
+      target="_blank"
+    >
+      {href}
+    </a>
+  );
+};
+
+const replaceLinks = (intialTextArray, matched, renderCallback) => {
+  // Construct an RegExp to capture all matched emails.
+  const regularExp = matched.reduce(
+    (prevVal, currVal, currentIndex) =>
+      currentIndex === 0 ? prevVal : `${prevVal}|(${escapeRegExp(currVal)})`,
+    `(${escapeRegExp(matched[0])})`,
+  );
+
+  // Split array items and flatten it.
+  const flatSplitArray = intialTextArray
+    .map(item => {
+      if (typeof item === 'string') {
+        return item
+          .split(new RegExp(regularExp, 'g'))
+          .filter(v => v !== undefined && v !== '');
```

Comment:
Describe in a comment what's going on there.

Reply:
I guess 
```suggestion
          .filter(v => !!v');
```
does the same thing.